src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerOptions.cs

```diff
@@ -102,146 +100,5 @@ public int MaxQueueLength
-            {
-                LogToStandardErrorThreshold = logToStandardErrorThreshold;
-            }
-
-            if (ParseInt(configuration, nameof(MaxQueueLength), out int maxQueueLength))
-            {
-                MaxQueueLength = maxQueueLength;
-            }
-
-            if (ParseEnum(configuration, nameof(QueueFullMode), out ConsoleLoggerQueueFullMode queueFullMode))
-            {
-                QueueFullMode = queueFullMode;
-            }
-
-#pragma warning disable CS0618 // Type or member is obsolete
-            if (configuration[nameof(TimestampFormat)] is string timestampFormat)
-            {
-                TimestampFormat = timestampFormat;
-            }
-#pragma warning restore CS0618 // Type or member is obsolete
-
-#pragma warning disable CS0618 // Type or member is obsolete
-            if (ParseBool(configuration, nameof(UseUtcTimestamp), out bool useUtcTimestamp))
-            {
-                UseUtcTimestamp = useUtcTimestamp;
-            }
-#pragma warning restore CS0618 // Type or member is obsolete
-        }
-
-        /// <summary>
-        /// Parses the configuration value at the specified key into a bool.
-        /// </summary>
-        /// <returns>true if the value was successfully found and parsed. false if the key wasn't found.</returns>
-        /// <exception cref="InvalidOperationException">Thrown when invalid data was found at the specified configuration key.</exception>
-        internal static bool ParseBool(IConfiguration configuration, string key, out bool value)
-        {
-            if (configuration[key] is string valueString)
-            {
-                try
-                {
-                    value = bool.Parse(valueString);
-                    return true;
-                }
-                catch (Exception e)
-                {
-                    ThrowInvalidConfigurationException(configuration, key, typeof(bool), e);
-                }
-            }
-
-            value = default;
-            return false;
-        }
-
-        /// <summary>
-        /// Parses the configuration value at the specified key into an enum.
-        /// </summary>
-        /// <returns>true if the value was successfully found and parsed. false if the key wasn't found.</returns>
-        /// <exception cref="InvalidOperationException">Thrown when invalid data was found at the specified configuration key.</exception>
-        internal static bool ParseEnum<T>(IConfiguration configuration, string key, out T value) where T : struct
-        {
-            if (configuration[key] is string valueString)
-            {
-                try
-                {
-                    value =
-#if NETFRAMEWORK || NETSTANDARD2_0
-                        (T)Enum.Parse(typeof(T), valueString, ignoreCase: true);
-#else
-                        Enum.Parse<T>(valueString, ignoreCase: true);
-#endif
```

Comment:
(_this comment is more general than this change. if we were to address it, it would be in a separate PR and this PR wouldn't need to block on it_)

Looking at your current generated code:

```C#
    public static ConsoleLoggerFormat ParseConsoleLoggerFormat(string value, Func<string?> getPath)
    {
        try
        {
            return (ConsoleLoggerFormat)Enum.Parse(typeof(ConsoleLoggerFormat), value, ignoreCase: true);
        }
        catch (Exception exception)
        {
            throw new InvalidOperationException($"Failed to convert configuration value at '{getPath()}' to type '{typeof(ConsoleLoggerFormat)}'.", exception);
        }
    }
```

There are a couple comments here:

1. Why do we need a new method for every enum type in the graph? Can't we write 1 general "ParseEnum" method, like this one?
2. We really should use `Enum.Parse<T>` when possible. I did a quick benchmark:

```C#
using BenchmarkDotNet.Attributes;

namespace Benchmark;

public enum ConsoleLoggerFormat
{
    Default,
    Systemd,
}

[MemoryDiagnoser]
public class EnumParseBenchmark
{
    [Benchmark]
    public ConsoleLoggerFormat ParseConsoleLoggerFormatGeneric() => ParseConsoleLoggerFormatGeneric("Systemd", () => "");
    [Benchmark]
    public ConsoleLoggerFormat ParseConsoleLoggerFormat() => ParseConsoleLoggerFormat("Systemd", () => "");

    public static ConsoleLoggerFormat ParseConsoleLoggerFormatGeneric(string value, Func<string?> getPath)
    {
        try
        {
            return Enum.Parse<ConsoleLoggerFormat>(value, ignoreCase: true);
        }
        catch (Exception exception)
        {
            throw new InvalidOperationException($"Failed to convert configuration value at '{getPath()}' to type '{typeof(ConsoleLoggerFormat)}'.", exception);
        }
    }

    public static ConsoleLoggerFormat ParseConsoleLoggerFormat(string value, Func<string?> getPath)
    {
        try
        {
            return (ConsoleLoggerFormat)Enum.Parse(typeof(ConsoleLoggerFormat), value, ignoreCase: true);
        }
        catch (Exception exception)
        {
            throw new InvalidOperationException($"Failed to convert configuration value at '{getPath()}' to type '{typeof(ConsoleLoggerFormat)}'.", exception);
        }
    }
}
```
on my machine I see:
|                          Method |     Mean |    Error |   StdDev |  Gen 0 | Allocated |
|-------------------------------- |---------:|---------:|---------:|-------:|----------:|
| ParseConsoleLoggerFormatGeneric | 27.92 ns | 0.855 ns | 0.985 ns |      - |         - |
|        ParseConsoleLoggerFormat | 72.91 ns | 0.270 ns | 0.226 ns | 0.0036 |      24 B |

The generic version is faster and doesn't allocate.

Reply:
Sounds good; will update in a follow-up.